precompile/allowlist/allowlist.abi

```diff
@@ -0,0 +1,73 @@
+[
+    {
+        "inputs": [
+            {
+                "internalType": "address",
+                "name": "addr",
+                "type": "address"
+            }
+        ],
+        "name": "readAllowList",
+        "outputs": [
+            {
+                "internalType": "uint256",
+                "name": "role",
+                "type": "uint256"
```

Comment:
is it possible to introduce an Enum?
is the `uint256` here the reason for using `big.Int` in packing?

Reply:
I will investigate if it's possible to introduce enum without breaking the contract. It might be possible. 

Yes, that might also be possible to convert to uint 8 etc. I will do that in another PR. but I probably won't upgrade if it breaks backward compatibility (even if we can upgrade with DUpgrade)